src/Core/Grid/Query/TitleQueryBuilder.php

```diff
@@ -0,0 +1,138 @@
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Grid\Query;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use PrestaShop\PrestaShop\Core\Grid\Search\SearchCriteriaInterface;
+
+/**
+ * Class TitleQueryBuilder provides query builders for titles grid.
+ */
+class TitleQueryBuilder extends AbstractDoctrineQueryBuilder
+{
+    /**
+     * @var DoctrineSearchCriteriaApplicatorInterface
+     */
+    private $searchCriteriaApplicator;
+
+    /**
+     * @var int
+     */
+    private $languageId;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param DoctrineSearchCriteriaApplicatorInterface $searchCriteriaApplicator
+     * @param int $languageId
+     */
+    public function __construct(
+        Connection $connection,
+        string $dbPrefix,
+        DoctrineSearchCriteriaApplicatorInterface $searchCriteriaApplicator,
+        int $languageId
+    ) {
+        parent::__construct($connection, $dbPrefix);
+
+        $this->searchCriteriaApplicator = $searchCriteriaApplicator;
+        $this->languageId = $languageId;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $builder = $this->getTitleQueryBuilder($searchCriteria)
+            ->select('t.*, tl.name');
+
+        $this->searchCriteriaApplicator
+            ->applySorting($searchCriteria, $builder)
+            ->applyPagination($searchCriteria, $builder);
+
+        return $builder;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        return $this->getTitleQueryBuilder($searchCriteria)->select('COUNT(t.id_gender)');
+    }
+
+    /**
+     * @param SearchCriteriaInterface $searchCriteria
+     *
+     * @return QueryBuilder
+     */
+    private function getTitleQueryBuilder(SearchCriteriaInterface $searchCriteria): QueryBuilder
+    {
+        $builder = $this->connection->createQueryBuilder()
+            ->from($this->dbPrefix . 'gender', 't')
+            ->innerJoin('t', $this->dbPrefix . 'gender_lang', 'tl', 't.id_gender = tl.id_gender')
+            ->andWhere('tl.`id_lang`= :language')
+            ->setParameter('language', $this->languageId)
+        ;
+
+        $this->applyFilters($builder, $searchCriteria);
+
+        return $builder;
+    }
+
+    /**
+     * @param QueryBuilder $builder
+     * @param SearchCriteriaInterface $searchCriteria
+     */
+    private function applyFilters(QueryBuilder $builder, SearchCriteriaInterface $searchCriteria): void
+    {
+        $allowedFilters = [
+            'id_gender',
+            'type',
+            'name',
+        ];
+
+        foreach ($searchCriteria->getFilters() as $filterName => $filterValue) {
+            if (!in_array($filterName, $allowedFilters)) {
+                continue;
+            }
+
+            if ($filterName === 'id_gender') {
```

Comment:
I think `type` should be tested as strict equality as well as it's a tinyint configuration

Reply:
Fixed